### PR TITLE
Add dateTo and dateFrom transaction filter

### PR DESCRIPTION
**Pull request by wandi34**
# Checklist for Definition of Done

Depending on the task some of the following jobs are not relevant. Please check the items or cross them out (`~~`text`~~`).

* [x] Code is technically reviewed by myself
* [x] Code is manually tested
* [x] Pipeline is green
* [x] Automatic tests for new functionality are created
* [x] Documentation is updated
* [x] Branch is rebased and therefore ready to merge
* [x] Git history is clean

### CODE REVIEW

File: service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/ais/AccountSpiImpl.java [hunk 1]
```diff
@@ -104,6 +104,7 @@ public AccountSpiImpl(TestDataService testDataService, TestDataMapper testDataMa
     }
```
Comment (atamanroman) [inline service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/ais/AccountSpiImpl.java:105]: Please remove logfile above and gitignore `service/logs`

File: service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/ais/AccountSpiImpl.java [hunk 2]
```diff
@@ -224,4 +225,8 @@ public AccountSpiImpl(TestDataService testDataService, TestDataMapper testDataMa
     return testDataService
         .getTransactions(psu.get().getPsuId(), accountId);
   }
+
+  private boolean dateIsInDateRange(LocalDate date, LocalDate from, LocalDate to) {
+    return date.compareTo(from) > 0 && date.compareTo(to) < 0;
```
Comment (atamanroman) [inline service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/ais/AccountSpiImpl.java]: `LocalDate#before/after` is easier to read.

**Review by atamanroman — CHANGES_REQUESTED** [review]

File: service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/ais/AccountSpiImpl.java (hunk 1 above)
Comment (wandi34) [reply service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/ais/AccountSpiImpl.java:105]: Done

File: service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/ais/AccountSpiImpl.java (hunk 2 above)
Comment (wandi34) [reply service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/ais/AccountSpiImpl.java]: Done

**Review by atamanroman — APPROVED** [review]

File: service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/ais/AccountSpiImpl.java [hunk 3]
```diff
@@ -224,4 +225,8 @@ public AccountSpiImpl(TestDataService testDataService, TestDataMapper testDataMa
     return testDataService
         .getTransactions(psu.get().getPsuId(), accountId);
   }
+
+  private boolean dateIsInDateRange(LocalDate date, LocalDate from, LocalDate to) {
+    return date.isAfter(from) && date.isBefore(to);
```
Comment (atamanroman) [inline service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/ais/AccountSpiImpl.java]: What about the edge cases? If `date == to` or `date == from` `date` should be included, right?

Comment (wandi34) [reply service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/ais/AccountSpiImpl.java]: Done

**Pull request merged**